File: scripts/isbn10.js
export default function check1(num){
    var arr=num.split(/-/)
    if (arr[3]=='X'){
        num=num.replace("X",10)
    }
    
    var id=+arr[3]
    var newnum=""
    for (var i=0;i<num.length;i++){
        if(num[i]!="-"){
            newnum+=num[i]
        }
    }
    var a=newnum.length
    if ((a==10) && (arr.length==4) && (arr[3]<10)){
        let sum=0
        var newnumb=Number(newnum.substring(0, newnum.length - 1))
        var fact=2
        while (newnumb>0){
            sum+=(newnumb%10)*fact
            newnumb=Math.floor(newnumb/10)
            fact++
        }
        if((sum+id)%11==0){
            return "ISBN-10";
        }
    }
    
}

File: C#/Isbn10.cs
using System;

namespace Is10
{
    public class Isbn10
    {
        public int Isb10(string numb)
        {
            string isbn10 = numb;
            string[] arr = isbn10.Split('-');
            if (arr[3] == "X")
                arr[3] = "10";
            isbn10 = isbn10.Replace("-", "");
            int id = Int32.Parse(arr[3]);
            int lenArr = arr.Length;
            int lenStr = isbn10.Length;
            if ((lenStr == 10) && (lenArr == 4) && (id < 10))
            {
                double sum = 0;
                double newnumb = Int32.Parse(isbn10.Substring(0, 9));
                int fact = 2;
                while (newnumb > 0)
                {
                    sum += (newnumb % 10) * fact;
                    newnumb = Math.Floor(newnumb / 10);
                    fact++;
                }
                if ((sum + id) % 11 == 0)
                {
                    Console.Write("ISBN-10");
                }
            }
            Console.ReadKey();
            return 1;
        }
    }
}